### PR TITLE
Fixed Freeze and removed warnings

### PR DESCRIPTION
**Pull request by SimonDanisch**
Did not test on 0.3, but `unsafe_convert` is in Compat, right?
Does this most crucial package actually have no test at all!?

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Culprit: JuliaLang/julia#9007

File: src/WinRPM.jl [hunk 1]
```diff
@@ -48,11 +48,11 @@ end
         if VERSION >= v"0.3-"
             res = ccall((:URLDownloadToCacheFileW,:urlmon),stdcall,Cuint,
               (Ptr{Void},Ptr{Uint16},Ptr{Uint16},Clong,Cint,Ptr{Void}),
-              0,utf16(source),dest,sizeof(dest)>>1,0,0)
+              C_NULL,utf16(source),dest,sizeof(dest)>>1,0,C_NULL)
         else
```
Comment (simonster) [inline src/WinRPM.jl]: It doesn't look like this branch should ever be taken, since `REQUIRE` specifies julia 0.3?

Comment (SimonDanisch) [inline src/WinRPM.jl]: True..unsafe_convert is not needed!

**Comment by SimonDanisch** [comment]
Oh so the same stuff happend over there:
JuliaLang/Homebrew.jl#90
JuliaLang/julia#9079

**Comment by tkelman** [comment]
Yikes, yeah 0.4 has a lot of recently-inflicted brokenness. WinRPM does tend to only get tested indirectly, but we can certainly add tests and set up appveyor here. Last time I tried to use WinRPM from Linux it wasn't able to do anything especially useful.

**Comment by vtjnash** [comment]
you should be able to use all of the functionality on linux by specifying an architecture string to the various functions

**Pull request merged**